src/ck-core/envelope.h

```diff
@@ -200,14 +200,9 @@ namespace ck {
   UShort epIdx;        /* Entry point to call */                               \
   ck::impl::s_attribs attribs;
 
-class envelope {
+class alignas(ALIGN_BYTES) envelope {
```

Comment:
I think this doesn't achieve the same effective alignment as before: 'alignas' here  in this PR will ensure that the envelope structure is laid out in memory beginning at a certain alignment, while the previous padding scheme was there to ensure that whatever structure follows this one (the user msg buffer) would be laid out in memory beginning at a certain alignment after the envelope. In other words, the padding was effectively there to ensure alignment not of the envelope itself but of what follows it.

The comments on this issue contain an illustration of the whole msg structure: https://github.com/UIUC-PPL/charm/issues/387

Reply:
I believe this should achieve the same effective alignment as before, `alignas` should both ensure that the envelope structure begins at a certain alignment _and_ that the size of the envelope structure is a multiple of the requested alignment, ergo the user message buffer will also start at the same alignment.

The way we use it, I think the property of beginning at a certain alignment is probably moot since we use things like `posix_memalign` to allocate memory without reading the alignment of the type.

Reply:
That's right, but I think there should be a comment added by the alignas statement clarifying that its purpose is really to align the user buffer that follows the envelope.

Separate from this PR, I think ALIGN_BYTES should receive a CMI_ prefix.

Reply:
> That's right, but I think there should be a comment added by the alignas statement clarifying that its purpose is really to align the user buffer that follows the envelope.
> 
> Separate from this PR, I think ALIGN_BYTES should receive a CMI_ prefix.

Ah, yes, I agree with both of those, good idea. I'll update this PR and create another to update `ALIGN_BYTES`.